### PR TITLE
[chore] implement shutdown verifier for metrics and logs

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: processor/processortest/shutdown_verifier.go [hunk 1]
```diff
@@ -44,9 +44,61 @@ func verifyTracesDoesNotProduceAfterShutdown(t *testing.T, factory processor.Fac
...
+func verifyLogsDoesNotProduceAfterShutdown(t *testing.T, factory processor.Factory, cfg component.Config) {
+	// Create a proc and output its produce to a sink.
+	nextSink := new(consumertest.LogsSink)
+	proc, err := factory.CreateLogsProcessor(context.Background(), NewNopCreateSettings(), cfg, nextSink)
+	if err != nil {
+		if errors.Is(err, component.ErrDataTypeIsNotSupported) {
+			return
+		}
+		require.NoError(t, err)
+	}
```
Comment (bogdandrutu) [inline processor/processortest/shutdown_verifier.go:56]:
```suggestion
	if errors.Is(err, component.ErrDataTypeIsNotSupported) {
		return
	}
	require.NoError(t, err)
```

File: processor/processortest/shutdown_verifier.go [hunk 2]
```diff
@@ -44,9 +44,61 @@ func verifyTracesDoesNotProduceAfterShutdown(t *testing.T, factory processor.Fac
...
+	// Now shutdown the proc.
+	assert.NoError(t, proc.Shutdown(context.Background()))
+
+	// The Shutdown() is done. It means the proc must have sent everything we
+	// gave it to the next sink.
+	assert.EqualValues(t, generatedCount*2, nextSink.DataPointCount())
+}
+
 // VerifyShutdown verifies the processor doesn't produce telemetry data after shutdown.
 func VerifyShutdown(t *testing.T, factory processor.Factory, cfg component.Config) {
```
Comment (bogdandrutu) [inline processor/processortest/shutdown_verifier.go:100]: You can write a test, that also can be used as an example for the nop processor.

**Review by bogdandrutu — APPROVED** [review]

File: processor/processortest/shutdown_verifier.go (hunk 2 above)
Comment (atoulme) [reply processor/processortest/shutdown_verifier.go:98]: sounds good

File: processor/processortest/shutdown_verifier.go (hunk 1 above)
Comment (bogdandrutu) [reply processor/processortest/shutdown_verifier.go]: @atoulme when I comment somewhere in your PR about a refactoring like this, I expect you to apply it everywhere else. Please, also you should do it for trace.

Comment (atoulme) [reply processor/processortest/shutdown_verifier.go]: sure thing!

File: processor/processortest/shutdown_verifier.go (hunk 2 above)
Comment (atoulme) [reply processor/processortest/shutdown_verifier.go:94]: test won't pass as nopprocessor doesn't pass on data - I suppose that's to be the expected behavior of the nopprocessor? Not sure. In the mean time I'll work on a simple processor accepting sinks.

**Review by codeboten — APPROVED** [review]

**Pull request merged**